### PR TITLE
Update to Rails 5.1

### LINKED ISSUE #614: DEPRECATION WARNING: Passing a class as a value in an Active Record query is deprecated
```
DEPRECATION WARNING: Passing a class as a value in an Active Record query is deprecated and will be removed. Pass a string instead. (called from block (3 levels) in <top (required)>
 at /home/travis/build/fatfreecrm/fat_free_crm/spec/lib/permissions_spec.rb:33)
```

Needs:

- [X] explicit coverage of ```remove_permissions```
- [X]  coverage of group, user class_eval generated methods that use group/user_ids- covered by permissions_spec

Not sure if foo.class.to_s is right vs foo.class.name

### PR DESCRIPTION
**Pull request by johnnyshields**
- Bump Rails version in gemspec
- Bundle update
- FieldGroup.tag needs to be an optional relationship
- Fix ActiveModelSerializers XML when using FFCRM as a gem (not sure if directly related to Rails 5)
- Fix #614

### CODE REVIEW

**Comment by johnnyshields** [comment]
This has one remaining error, not exactly sure but it looks like the specs are trying to assign invalid user_ids. Would appreciate some help on this if anybody knows the reason.

```
  1) FatFreeCRM::Permissions user_ids should assign permissions to the object
     
Failure/Error: @entity.user_ids = %w(1 2 3)

TypeError:
  can't cast Class
```

**Comment by CloCkWeRX** [comment]
I reckon this is fine, but I might split it into a few cleaner commits - bump rails, fix alias_method_chain, version activerecord migrations, etc.

File: lib/fat_free_crm/core_ext/alias_method_chain.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class Module
+
+  def alias_method_chain(target, feature)
```
Comment (CloCkWeRX) [inline lib/fat_free_crm/core_ext/alias_method_chain.rb]: Potentially not needed anymore with the swap to prepend

**Comment by CloCkWeRX** [comment]
I reckon this is good to go, once we kill off the alias_chain_method stuff - the other minor syntax cleanup is small enough/digestable enough that we can pinpoint any issues that crop up.

I'll dig up some of the other tickets this closes off too.

File: lib/fat_free_crm/core_ext/alias_method_chain.rb (hunk 1 above)
Comment (johnnyshields) [reply lib/fat_free_crm/core_ext/alias_method_chain.rb]: @CloCkWeRX this is needed due to gem dependencies which are not yet using the prepend syntax. Build will fail without it. I have converted all FFCRM code to use prepend.

**Comment by CloCkWeRX** [comment]
Sigh, just realised I hadn't cut a stable 0.15.0 release with rails 5.0.0. Will do that, merge this in, do a 0.16.0 release

File: lib/fat_free_crm/core_ext/alias_method_chain.rb (hunk 1 above)
Comment (CloCkWeRX) [reply lib/fat_free_crm/core_ext/alias_method_chain.rb]: Lets ticket for removal in 0.16.0 then, and maybe add some deprecation warnings like http://api.rubyonrails.org/v5.0/classes/ActiveSupport/Deprecation/Reporting.html lets you do

File: lib/fat_free_crm/permissions.rb [hunk 2]
```diff
@@ -44,7 +44,7 @@ def #{model}_ids=(value)
               permissions_to_remove = Permission.where(
                 #{model}_id: self.#{model}_ids - value,
                 asset_id: self.id,
-                asset_type: self.class
+                asset_type: self.class.to_s
```
Comment (CloCkWeRX) [inline lib/fat_free_crm/permissions.rb]: Should this be to_s or class_name? I'm guessing it magically worked out as class_name previously. Lets split out and look at tests, so we can close #614 in isolation

Comment (CloCkWeRX) [reply lib/fat_free_crm/permissions.rb]: Nevermind, it's on a Class not an Object so boils down to the same thing.

File: lib/fat_free_crm/core_ext/alias_method_chain.rb (hunk 1 above)
Comment (johnnyshields) [reply lib/fat_free_crm/core_ext/alias_method_chain.rb]: @CloCkWeRX removal will not be possible unless we remove all alias_method_chain usage from all third-party gem dependencies, which we do not control. Unfortunately this patch will be here for quite awhile.

Comment (CloCkWeRX) [reply lib/fat_free_crm/core_ext/alias_method_chain.rb]: Semver will save us, even if it means that we cut off a bunch of older gems that aren't 0.16.0+ compatible.

Comment (johnnyshields) [reply lib/fat_free_crm/core_ext/alias_method_chain.rb]: @CloCkWeRX we should raise a separate issue to do that and not make it a blocker for the 0.16.0 release. We cannot guarantee the timing at which other gem owners will release a version that removes `alias_method_chain`

**Comment by CloCkWeRX** [comment]
I'll bump master up to more modern rubocop shortly, part of getting the other style changes all landed; which should fix the build error

**Comment by johnnyshields** [comment]
sure yeah we will need to run `rubocop --auto-gen-config again`

**Comment by johnnyshields** [comment]
@CloCkWeRX this is good to merge

**Comment by CloCkWeRX** [comment]
Let's raise a seperate one for killing off alias_method_chain

**Pull request merged**